expression/builtin_cast.go

```diff
@@ -1364,7 +1364,7 @@ func WrapWithCastAsReal(ctx context.Context, expr Expression) Expression {
 	if expr.GetType().EvalType() == types.ETReal {
 		return expr
 	}
-	tp := types.NewFieldType(mysql.TypeDouble)
+	tp := types.ConstMySQLDouble
```

Comment:
A constant must not be modified.
It's obvious if the modification near the assignment like this, but not if it's modified somewhere else.